Robinhood/Robinhood.py

```diff
@@ -174,81 +218,94 @@ def investment_profile(self):
         res = res.json()
 
         # if requesting all, return entire object so may paginate with ['next']
-        if (stock == ""):
+        # Not sure variable returns types here is the best approach..
+        # API doesn't return pagination though when query is non-empty query=a ??
+        #if query is None and not (symbol or instrumentid):
+        #     return res
+        # XXX perhaps should return an iterable to hide the pagination, e.g. res['next'], res['previous'] aspects
+        if 'results' not in res:
             return res
-
         return res['results']
 
+    def instrument_splits(self, instrumentid=None):
+        """fetch instruments splits endpoint
 
-    def instrument(self, id):
-        """Fetch instrument info
+        Args:
+            instrumentid (str): instrumentid [uuid without the rest of URL]
 
-            Args:
-                id (str): instrument id
+        Returns:
+            (:obj:`dict`): JSON contents from `instruments splits` endpoint
 
-            Returns:
-                (:obj:`dict`): JSON dict of instrument
         """
-        url = str(self.endpoints['instruments']) + str(id) + "/"
-
-        try:
-            req = requests.get(url)
-            req.raise_for_status()
-            data = req.json()
-        except requests.exceptions.HTTPError:
-            raise RH_exception.InvalidInstrumentId()
+        res = None
+        if instrumentid:
+            res = self.session.get(
+                self.endpoints['instrumentsplits'].format(instrumentid=instrumentid)
+            )
+            self.raiseForStatus(res)
+            return res.json()
+        raise ValueError("Invalid instrumentid passed")
+
+    def quote_data(self, stock):
+        """fetch single stock quote
 
-        return data
-
-
-    def quote_data(self, stock=''):
-        """Fetch stock quote
+        Args:
+            stock (str): stock ticker, prompt if blank
 
             Args:
-                stock (str): stock ticker, prompt if blank
+                id (str): instrument id
 
             Returns:
-                (:obj:`dict`): JSON contents from `quotes` endpoint
+                (:obj:`dict`): JSON dict of instrument
         """
+        url = str(self.endpoints['quotes']) + str(stock) + "/"
 
-        url = None
-
-        if stock.find(',') == -1:
-            url = str(self.endpoints['quotes']) + str(stock) + "/"
-        else:
-            url = str(self.endpoints['quotes']) + "?symbols=" + str(stock)
-
-        #Check for validity of symbol
+        # Check for validity of symbol
         try:
-            req = requests.get(url)
-            req.raise_for_status()
-            data = req.json()
+            res = requests.get(url)
+            self.raiseForStatus(res)
         except requests.exceptions.HTTPError:
-            raise RH_exception.InvalidTickerSymbol()
-
+            raise self.exceptions.InvalidTickerSymbol()
 
-        return data
+        return res.json()
```

Comment:
`res.json()` should remain inside try/except b/c it is possible to return HTML/XML if a particularly heinous error happens on RH's side... the `raise_for_status` _should_ catch 500 errs, but json parse is still possible to flub from RH's side